source/Ship.cpp

```diff
@@ -2306,6 +2306,16 @@ void Ship::DoGeneration()
 		shieldDelay = max(0, shieldDelay - 1);
 		hullDelay = max(0, hullDelay - 1);
 	}
+	// Carried ships can steal energy from their parent's batteries,
+	// if they are preparing for deployment. Otherwise, they replenish the
+	// parent's batteries.
+	if(!GetSystem())
```

Comment:
Is this correct? When you're *not* in a system?

Reply:
When you are carried by a ship.

Reply:
yea, should be the check for being carried

Reply:
Ooh right